lib/toJsonKey.js

```diff
@@ -0,0 +1,48 @@
+"use strict";
+
+/**
+ * Helper for getting a reproducible, key-sorted string representation of object.
+ * Used to interpret an object as a falcor key.
+ * @function
+ * @param {Object} obj
+ * @return stringified object with sorted keys.
+ */
+function toJsonKey(obj) {
+    if (typeof obj !== "object" ||
+        obj === null ||
+        Array.isArray(obj)) {
+        return JSON.stringify(obj);
+    }
```

Comment:
Should we throw a TypeError here since `isJsonKey(result)` will return false?

Reply:
yes, throwing a TypeError in these cases now, and also throwing a TypeError for other cases such as `Date`, `Map`, `WeakSet`, etc, because `isJsonKey()` will return false for them as well